### PR TITLE
[Bug fix] Metrics datasource

### PR DESCRIPTION
**Pull request by TackAdam**
### Description
While datasource is enabled, set initial condition to null and wait to render page until the selection/default is registered.

Currently if a localhost is not defined the call will trigger the search on a non valid datasource.
NoLocalHost

This change prevents the search call from being triggered until the datasource has been read.

https://github.com/user-attachments/assets/75e8559e-07dc-46d3-b9bf-fe127f83ee65

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/metrics/sidebar/sidebar.tsx [hunk 1]
```diff
@@ -95,6 +97,11 @@ export const Sidebar = ({
   }, [selectedMetrics, selectedMetricsIds, dataSourceMDSId]);
 
   useEffect(() => {
+    // If data source is enabled but is still null prevent the invalid call
+    if (dataSourceEnabled && dataSourceMDSId === null) {
```
Comment (joshuali925) [inline public/components/metrics/sidebar/sidebar.tsx:101]: if local cluster is selected, what would `dataSourceMDSId` be?

**Review by joshuali925 — APPROVED** [review]

File: public/components/metrics/sidebar/sidebar.tsx (hunk 1 above)
Comment (TackAdam) [reply public/components/metrics/sidebar/sidebar.tsx:101]: Local cluster is just an empty string "" when selected it get sets and is no longer null

**Review by ps48 — APPROVED** [review]

File: public/components/metrics/index.tsx [hunk 2]
```diff
@@ -77,7 +77,7 @@ export const Home = ({
   const dispatch = useDispatch();
 
   const onSelectedDataSource = async (dataSources: DataSourceOption[]) => {
-    const id = dataSources[0] ? dataSources[0].id : '';
+    const id = dataSources[0] ? dataSources[0].id : null;
```
Comment (ps48) [inline public/components/metrics/index.tsx:80]: Only one question: What's the condition when `dataSources[0]` will be undefined and we have to set the id as `null`?

Comment (TackAdam) [reply public/components/metrics/index.tsx:80]: The only condition this should be triggered is when a workspace is created and its datasource is removed putting it in a state where it has no data sources available at all. In this case metrics won't work and we don't want to display options. 
We should probably have an empty state for this such as " No connected data source with redirection to associate one".

**Pull request merged**